### PR TITLE
Big file support for interpreter

### PR DESCRIPTION
**Pull request by MatiBrego**
- Refactored interpreter to receive AST and Environment, and return log outputs and new Environment
- Made Environment immutable
- Added statement interpreter for single statement AST nodes.

Note: merge #20 before checking this one.

### CODE REVIEW

File: interpreter/src/main/kotlin/interpreter/ProgramInterpreter.kt
Comment (JuanSeZ) [inline interpreter/src/main/kotlin/interpreter/ProgramInterpreter.kt:1]: When we will use the Program Interpreter?

File: interpreter/src/main/kotlin/interpreter/StatementInterpreter.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package interpreter
+
+import ast.AST
+import kotlin.reflect.KClass
+
+class StatementInterpreter(
+    private val handlers: Map<KClass<out AST>, Interpreter>,
+) : Interpreter {
```
Comment (JuanSeZ) [inline interpreter/src/main/kotlin/interpreter/StatementInterpreter.kt:8]: 🤤

File: interpreter/src/main/kotlin/interpreter/ProgramInterpreter.kt
Comment (MatiBrego) [reply interpreter/src/main/kotlin/interpreter/ProgramInterpreter.kt:1]: The idea was to have backwards compatibility until we remove all of its uses.

File: interpreter/src/main/kotlin/interpreter/expression/OperationInterpreter.kt
Comment (JuanSeZ) [inline interpreter/src/main/kotlin/interpreter/expression/OperationInterpreter.kt:1]: You'll probably want to handle differently cause of #22. Maybe separate into Binary and Unary

File: parser/build.gradle [hunk 2]
```diff
@@ -4,5 +4,7 @@ plugins {
 
 dependencies {
     implementation project(':common')
+    testImplementation project(':lexer')
```
Comment (JuanSeZ) [inline parser/build.gradle:7]: Idk how I feel about this

Comment (JuanSeZ) [reply parser/build.gradle:7]: nah I like it

**Review by JuanSeZ — APPROVED** [review]
LGTM

**Comment by JuanSeZ** [comment]
Good job!

**Pull request merged**